code/modules/supermatter/supermatter.dm

```diff
@@ -415,16 +415,22 @@
 
 	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, 1)
 
+	if(istype(AM, /obj/item/weapon/holder))
+		var/obj/item/weapon/holder/H = AM
+		if(H.stored_mob)
+			Consume(H.stored_mob)
 	Consume(AM)
 
-
 /obj/machinery/power/supermatter/proc/Consume(var/mob/living/user)
 	if(istype(user))
 		user.dust()
-		if(istype(user,/mob/living/simple_animal/mouse)) //>implying mice are going to follow the rules
+		if(ismouse(user)) //>implying mice are going to follow the rules
 			return
 		power += 200
 	else
+		if(istype(user, /obj/item/weapon/holder)) //STOP MAKING POWER OUT OF HOLDERS HOLY SHIT
+			qdel(user)
+			return
```

Comment:
Immediately after this block, it calls qdel anyway.  There should be a istypeinlist() check afterward that which returns on match, or sets the power modifier from 200 to 0, or whatever tickles your fancy.